Bottom: 5a760faafcc983b0685ff19c9c9ed318d1719da7
Top:    507bc04297fef02583815456d8401f5930fe1e22
Author: Ben Greear <greearb@candelatech.com>
Date:   2014-02-03 12:18:38 -0800

cifs: Authentication patch from shirishpargaonkar@gmail.com

Might help with problem netapp saw, but not certain at this point.

Signed-off-by: Ben Greear <greearb@candelatech.com>


---

diff --git a/fs/cifs/sess.c b/fs/cifs/sess.c
index af0ec2d..fe84acb 100644
--- a/fs/cifs/sess.c
+++ b/fs/cifs/sess.c
@@ -459,9 +459,8 @@ int build_ntlmssp_auth_blob(unsigned char *pbuffer,
 	sec_blob->WorkstationName.MaximumLength = 0;
 	tmp += 2;
 
-	if (((ses->ntlmssp->server_flags & NTLMSSP_NEGOTIATE_KEY_XCH) ||
-		(ses->ntlmssp->server_flags & NTLMSSP_NEGOTIATE_EXTENDED_SEC))
-			&& !calc_seckey(ses)) {
+	if ((ses->ntlmssp->server_flags & NTLMSSP_NEGOTIATE_KEY_XCH) &&
+	    !calc_seckey(ses)) {
 		memcpy(tmp, ses->ntlmssp->ciphertext, CIFS_CPHTXT_SIZE);
 		sec_blob->SessionKey.BufferOffset = cpu_to_le32(tmp - pbuffer);
 		sec_blob->SessionKey.Length = cpu_to_le16(CIFS_CPHTXT_SIZE);
